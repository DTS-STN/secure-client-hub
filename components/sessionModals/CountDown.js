import { useState, useEffect } from 'react'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { icon } from '../../lib/loadIcons'
import { Button } from '@dts-stn/service-canada-design-system'
import { solid } from '@fortawesome/fontawesome-svg-core/import.macro'

const Countdown = (props) => {
  const [clock, setClock] = useState({
    days: 0,
    hours: 0,
    minutes: 0,
    seconds: 0,
  })
  const { days, hours, minutes, seconds } = clock

  const getTime = () => {
    const time = Date.parse(props.deadline) - Date.now()

    setClock((prev) => {
      return {
        ...prev,
        days: Math.floor(time / (1000 * 60 * 60 * 24)),
        hours: Math.floor((time / (1000 * 60 * 60)) % 24),
        minutes: Math.floor((time / 1000 / 60) % 60),
        seconds: Math.floor((time / 1000) % 60),
      }
    })
  }

  useEffect(() => {
    const interval = setInterval(() => getTime(props.deadline), 1000)

    return () => clearInterval(interval)
  }, [])

  return (
    <div
      className="m-8 sm:mx-24 sm:mt-24 p-4 md:p-16 bg-white rounded h-fit"
      data-cy="exitBetaModal"
      id={props.id}
    >
      <div className="flex justify-between pb-5">
        <div
          className="text-3xl font-display font-bold"
          role="heading"
          aria-level="1"
        >
          Stay signed in?
        </div>
        <button
          data-cy="x-button"
          type="button"
          aria-label={'Close Modal'}
          onClick={props.closeModal}
        >
          <FontAwesomeIcon aria-hidden="true" icon={solid('xmark')} size="xl" />
        </button>
      </div>

      <div className="flex">
        <div className="flex-none flex items-top justify-center pr-3">
          <FontAwesomeIcon
            icon={icon['triangle-exclamation']}
            size="lg"
            color="orange"
          />
        </div>
        <div className="flex-auto">
          <p className="mr-6">
            You have been active for a while so we will sign you out to keep
            your information secure.
          </p>
          <p className="font-bold mr-6">
            You will be signed out in {minutes} minutes and {seconds}
            seconds.
          </p>
        </div>
      </div>

      <div className="flex pt-10">
        <Button
          text="Sign Out"
          onClick={props.onSignOut}
          styling="secondary"
          className="mr-3"
        />
        <Button
          text="Stay Signed In"
          onClick={props.onStay}
          styling="primary"
          className="mr-3"
        />
      </div>
    </div>
  )
}

export default Countdown
