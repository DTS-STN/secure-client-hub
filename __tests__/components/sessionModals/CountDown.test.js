import renderer from 'react-test-renderer'
import CountDown from '../../../components/sessionModals/CountDown'

it('renders correctly', () => {
  const countDownRender = renderer
    .create(
      <CountDown
        closeModal={() => console.log('Close Modal')}
        onSignOut={() => console.log('Sign Out Clicked')}
        onStay={() => console.log('Stay Signed In Clicked')}
        id="CountDown"
        deadline="January, 31, 2023"
      />
    )
    .toJSON()
  expect(countDownRender).toMatchSnapshot()
})
