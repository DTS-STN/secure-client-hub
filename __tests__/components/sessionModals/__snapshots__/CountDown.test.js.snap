// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="m-8 sm:mx-24 sm:mt-24 p-4 md:p-16 bg-white rounded h-fit"
  data-cy="exitBetaModal"
  id="CountDown"
>
  <div
    className="flex justify-between pb-5"
  >
    <div
      aria-level="1"
      className="text-3xl font-display font-bold"
      role="heading"
    >
      Stay signed in?
    </div>
    <button
      aria-label="Close Modal"
      data-cy="x-button"
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-xmark fa-xl "
        data-icon="xmark"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M310.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L160 210.7 54.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L114.7 256 9.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L160 301.3 265.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L205.3 256 310.6 150.6z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
  </div>
  <div
    className="flex"
  >
    <div
      className="flex-none flex items-top justify-center pr-3"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-triangle-exclamation fa-lg "
        color="orange"
        data-icon="triangle-exclamation"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 32c14.2 0 27.3 7.5 34.5 19.8l216 368c7.3 12.4 7.3 27.7 .2 40.1S486.3 480 472 480H40c-14.3 0-27.6-7.7-34.7-20.1s-7-27.8 .2-40.1l216-368C228.7 39.5 241.8 32 256 32zm0 128c-13.3 0-24 10.7-24 24V296c0 13.3 10.7 24 24 24s24-10.7 24-24V184c0-13.3-10.7-24-24-24zm32 224c0-17.7-14.3-32-32-32s-32 14.3-32 32s14.3 32 32 32s32-14.3 32-32z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
    <div
      className="flex-auto"
    >
      <p
        className="mr-6"
      >
        You have been active for a while so we will sign you out to keep your information secure.
      </p>
      <p
        className="font-bold mr-6"
      >
        You will be signed out in 
        0
         minutes and 
        0
        seconds.
      </p>
    </div>
  </div>
  <div
    className="flex pt-10"
  >
    <button
      className="ds-flex ds-flex-row ds-btn-secondary focus:ds-ring focus:ds-ring-offset-4 mr-3 "
      id="btn1"
      onClick={[Function]}
    >
      Sign Out
    </button>
    <button
      className="ds-flex ds-flex-row ds-btn-primary focus:ds-ring focus:ds-ring-offset-4 mr-3 "
      id="btn1"
      onClick={[Function]}
    >
      Stay Signed In
    </button>
  </div>
</div>
`;
